homeassistant/components/media_player/plex.py

```diff
@@ -0,0 +1,199 @@
+"""
+homeassistant.components.media_player.plex
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Provides an interface to the Plex API
+
+Configuration:
+
+To use Plex add something like this to your configuration:
+
+media_player:
+  platform: plex
+  name: plex
+  user: plex
+  password: my_secure_password
+
+Variables:
+
+name
+*Required
+The name of the backend device
+
+device
+*Required
+The frontend device (Look under settings > devices).
+
+user
+*Required
+The Plex username
+
+password
+*Required
+The Plex password
+"""
+
+import logging
+
+from homeassistant.components.media_player import (
+    MediaPlayerDevice, SUPPORT_PAUSE, SUPPORT_PREVIOUS_TRACK,
+    SUPPORT_NEXT_TRACK, MEDIA_TYPE_TVSHOW, MEDIA_TYPE_VIDEO)
+from homeassistant.const import (
+    STATE_IDLE, STATE_PLAYING, STATE_PAUSED, STATE_UNKNOWN)
+from plexapi.myplex import MyPlexUser
+
+REQUIREMENTS = ['https://github.com/miniconfig/python-plexapi/archive/'
+                '437e36dca3b7780dc0cb73941d662302c0cd2fa9.zip'
+                '#python-plexapi==1.0.2.5']
+
+_LOGGER = logging.getLogger(__name__)
+
+SUPPORT_PLEX = SUPPORT_PAUSE | SUPPORT_PREVIOUS_TRACK | SUPPORT_NEXT_TRACK
+
+# pylint: disable=abstract-method
+# pylint: disable=unused-argument
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """ Sets up the plex platform. """
+    name = config.get('name', '')
+    user = config.get('user', '')
+    password = config.get('password', '')
+    plexuser = MyPlexUser.signin(user, password)
+    plexserver = plexuser.getResource(name).connect()
+    dev = plexserver.clients()
+    for device in dev:
+        if "PlayStation" not in device.name:
+            add_devices([PlexClient(device.name, plexserver)])
+
+
+class PlexClient(MediaPlayerDevice):
+    """ Represents a Plex device. """
+
+    # pylint: disable=too-many-public-methods
+
+    def __init__(self, name, plexserver):
+        self.client = plexserver.client(name)
+        self._name = name
+        self._media = None
+        self.update()
+        self.server = plexserver
+
+    @property
+    def name(self):
+        """ Returns the name of the device. """
+        return self._name
+
+    @property
+    def state(self):
+        """ Returns the state of the device. """
+        if self._media is None:
+            return STATE_IDLE
+        else:
+            state = self._media.get('state')
+            if state == 'playing':
+                return STATE_PLAYING
+            elif state == 'paused':
+                return STATE_PAUSED
+            else:
+                return STATE_UNKNOWN
```

Comment:
This is really more of a stylistic suggestion than anything, but please always leave a default "fall through return" for things like this. It makes for a lot less bugs in the future when more stuff might be added to this or this code might get refactored. ie:

```python
def state(self):
        if self._media is None:
            return STATE_IDLE
        else:
            state = self._media.get('state')
            if state == 'playing':
                return STATE_PLAYING
            elif state == 'paused':
                return STATE_PAUSED
    return STATE_UNKNOWN
```